src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -765,4 +776,113 @@ public void validateScoreSettings(Exercise exercise) {
+
+        if (!feedbackList.isEmpty()) {
+            exercise.setGradingInstructionFeedbackUsed(true);
+        }
+    }
+
+    /**
+     * Re-evaluates the exercise before saving
+     * 1. The feedback associated with the exercise grading instruction needs to be updated
+     * 2. After updating feedbacks, result needs to be re-calculated
+     *
+     * @param exercise exercise to re-evaluate
+     * @param deleteFeedbacks  about checking if the feedbacks should be deleted when the associated grading instructions are deleted
+     *
+     */
+    public void reEvaluateExercise(Exercise exercise, Boolean deleteFeedbacks) {
+
+        List<GradingCriterion> gradingCriteria = exercise.getGradingCriteria();
+        // retrieve the feedbacks associated with the grading instructions
+        List<Feedback> feedbackList = feedbackRepository.findFeedbackByStructuredGradingInstructionId(gradingCriteria);
+
+        // collect each sub-grading instructions into the list
+        List<GradingInstruction> instructionList = gradingCriteria.stream().flatMap(gradingCriterion -> gradingCriterion.getStructuredGradingInstructions().stream())
+                .collect(Collectors.toList());
+
+        // update the related fields for feedbacks
+        for (GradingInstruction instruction : instructionList) {
+            for (Feedback feedback : feedbackList) {
+                if (feedback.getGradingInstruction().getId().equals(instruction.getId())) {
+                    feedback.setCredits(instruction.getCredits());
+                    feedback.setPositive(feedback.getCredits() >= 0);
+                    feedback.setDetailText(instruction.getFeedback());
+                }
+            }
+        }
+        feedbackRepository.saveAll(feedbackList);
+
+        List<Feedback> feedbacksDeleted = getFeedbacksShouldDeleteAfterSGIChanged(deleteFeedbacks, instructionList, exercise);
+
+        List<Result> results = resultRepository.findWithEagerSubmissionAndFeedbackByParticipationExerciseId(exercise.getId());
+
+        // re-calculate the results after updating the feedbacks
+        for (Result result : results) {
+            if (!feedbacksDeleted.isEmpty()) {
+                List<Feedback> existingFeedbacks = result.getFeedbacks();
+                if (!existingFeedbacks.isEmpty()) {
+                    existingFeedbacks.removeAll(feedbacksDeleted);
+                }
+                // first save the feedback (that is not yet in the database) to prevent null index exception
+                var savedFeedbackList = feedbackRepository.saveFeedbacks(existingFeedbacks);
+                result.updateAllFeedbackItems(savedFeedbackList, exercise instanceof ProgrammingExercise);
+            }
+
+            if (!(exercise instanceof ProgrammingExercise)) {
+                resultRepository.submitResult(result, exercise);
+            }
+            else {
+                double points = programmingAssessmentService.calculateTotalScore(result);
```

Comment:
I would call it `totalScore` as this is more intuitive when calling `calculateTotalScore`